### PR TITLE
[WIP] Move unit tests which were in integration tests - More py3 and mocking fixes

### PR DESCRIPTION
**Pull request by s0undt3ch**
This needs to be merged after https://github.com/saltstack/salt/pull/40138 is merged

### CODE REVIEW

**Review by cachedout — CHANGES_REQUESTED** [review]
Just one lint error: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/9957/violations/file/tests/unit/utils/test_docker.py/

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Review by cachedout — APPROVED** [review]

**Comment by s0undt3ch** [comment]
@cachedout this should be ready for you to take another look

File: tests/unit/modules/test_file.py [hunk 1]
```diff
@@ -759,17 +761,89 @@ def setup_loader_modules(self):
...
+            ret = filemod.source_list(
+                [{self.myfile: ''}], 'filehash', 'base')
+            self.assertEqual(list(ret), [self.myfile, 'filehash'])
+
+    def test_source_list_for_list_returns_local_file_proto_from_dict(self):
+        with patch.dict(filemod.__salt__, {'cp.list_master': MagicMock(return_value=[]),
+                                           'cp.list_master_dirs': MagicMock(return_value=[])}):
+            ret = filemod.source_list(
+                [{'file://' + self.myfile: ''}], 'filehash', 'base')
+            print(8888, ret)
```
Comment (s0undt3ch) [inline tests/unit/modules/test_file.py]: Bah!

File: tests/unit/netapi/rest_tornado/test_handlers.py [hunk 2]
```diff
@@ -3,27 +3,28 @@
...
 import hashlib
 
 # Import Salt Testing Libs
-from tests.support.unit import skipIf
-import tests.integration as integration
+from tests.integration import AdaptedConfigurationTestCaseMixIn
+from tests.support.unit import TestCase, skipIf
 
-# Import Salt libs
+# Import Salt libse
```
Comment (s0undt3ch) [inline tests/unit/netapi/rest_tornado/test_handlers.py]: Fix comment

**Comment by s0undt3ch** [comment]
Fixed my own review comments.

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by s0undt3ch** [comment]
Cachim!

**Comment by s0undt3ch** [comment]
@cachedout this is ready for a final review. No more commits will be added to this PR.

**Review by cachedout — APPROVED** [review]

**Comment by s0undt3ch** [comment]
Woohoo!!!! Gimme a few mins for another one like this :)

**Pull request merged**